### PR TITLE
eswords now have a demoliton_mod of 1.5x

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

most energy sword-like weapons have had their demolition_mod increased to 1.5x (from 1x). energy scalpels did not receive this buff, as they're easily crew-accessible and I don't want to deal with a balance flamewar.

this affects damage vs. robots, objects, structures, etc.

the melee part of the mecha damage unit test now accounts for demolition_mod.

## Why It's Good For The Game

image

## Changelog

:cl: ATHATH
balance: Most energy sword-like weapons have had their demolition_mod increased to 1.5x (from 1x). This affects damage vs. robots, objects, structures, etc. Energy scalpels did not receive this buff.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
You know what? Unironically great idea.

File: code/game/objects/items/dualsaber.dm [hunk 1]
```diff
@@ -29,6 +29,7 @@
 	resistance_flags = FIRE_PROOF
 	wound_bonus = -10
 	bare_wound_bonus = 20
+	demolition_mod = 1.5 //1.5x damage to objects, robots, etc.
```
Comment (tralezab) [inline code/game/objects/items/dualsaber.dm:32]: Shouldn't the base variable declaration just describe what the demolition mod does itself

File: code/modules/unit_tests/mecha_damage.dm [hunk 2]
```diff
@@ -25,7 +25,7 @@
 	// Get a sample "melee" weapon.
 	// The energy axe is chosen here due to having a high base force, to make sure we get over the equipment DT.
 	var/obj/item/dummy_melee = allocate(/obj/item/melee/energy/axe)
-	var/expected_melee_damage = round(dummy_melee.force * (1 - expected_melee_armor / 100), DAMAGE_PRECISION)
+	var/expected_melee_damage = round(dummy_melee.force * (1 - expected_melee_armor / 100) * dummy_melee.demolition_mod, DAMAGE_PRECISION)
```
Comment (tralezab) [inline code/modules/unit_tests/mecha_damage.dm:28]: Just mention you did this in the pr body, I'll add the fix tag

**Review by tralezab — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/mecha_damage.dm (hunk 2 above)
Comment (ATH1909) [reply code/modules/unit_tests/mecha_damage.dm:28]: :D

File: code/game/objects/items/dualsaber.dm (hunk 1 above)
Comment (ATH1909) [reply code/game/objects/items/dualsaber.dm:32]: probably, but the redundancy can't hurt, right?

**Comment by Pinkufeck** [comment]
May i ask what demolition_mod does

**Comment by moocowswag** [comment]
If it applies to silicons, why not augmented limbs (literally the same parts) and androids.

**Comment by davethwave** [comment]
Because the demolition modifier doesn't affect those things. This would require a lot more than just changing a 1 to a 1.5.

**Pull request merged**